File: ColocKitchenRace/Models/CohouseType.swift
//
//  CohouseType.swift
//  ColocKitchenRace
//
//  Created by Julien Rahier on 11/02/2026.
//

import Foundation

enum CohouseType: String, Codable, CaseIterable, Identifiable, Hashable {
    case mixed = "mixed"
    case girls = "girls"
    case boys = "boys"

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .mixed: return "Mixte"
        case .girls: return "Filles"
        case .boys: return "Garçons"
        }
    }
}

File: android/app/src/main/java/dev/rahier/colockitchenrace/data/model/CohouseType.kt
package dev.rahier.colockitchenrace.data.model

enum class CohouseType(val displayName: String) {
    MIXED("Mixte"),
    GIRLS("Filles"),
    BOYS("Garcons");

    companion object {
        fun fromFirestore(value: String): CohouseType? = when (value) {
            "mixed" -> MIXED
            "girls" -> GIRLS
            "boys" -> BOYS
            else -> null
        }
    }

    fun toFirestore(): String = when (this) {
        MIXED -> "mixed"
        GIRLS -> "girls"
        BOYS -> "boys"
    }
}
